File: Difficulty: Medium/Make Matrix Beautiful/make-matrix-beautiful.py
class Solution:
    def balanceSums(self, mat):
        n = len(mat)
        if n == 0:
           return 0

        max_row_sum = 0
        for i in range(n):
            current_row_sum = 0
            for j in range(n):
                current_row_sum += mat[i][j]

            if current_row_sum > max_row_sum:
                max_row_sum = current_row_sum

        max_col_sum = 0
        for j in range(n):
            current_col_sum = 0
            for i in range(n):
                current_col_sum += mat[i][j]

            if current_col_sum > max_col_sum:
                max_col_sum = current_col_sum

        target_sum = max(max_row_sum, max_col_sum)

        total_operations = 0
        for i in range(n):
            current_row_sum = 0
            for j in range(n):
                current_row_sum += mat[i][j]

            total_operations += (target_sum - current_row_sum)

        return total_operations

File: Difficulty: Medium/Make Matrix Beautiful/make-matrix-beautiful.cpp
class Solution {
  public:
    int balanceSums(vector<vector<int>>& mat) {
        // code here
        int max_sum=0;
        int n = mat.size();
        int m = mat[0].size();

        vector<int> row_sum(n, 0);
        vector<int> col_sum(m, 0);
        for(int i=0;i<mat.size();i++){
            for(int j=0;j<mat[0].size();j++){
                row_sum[i]+=mat[i][j];
            }
            max_sum=max(max_sum,row_sum[i]);
        }
        for(int i=0;i<mat.size();i++){
            for(int j=0;j<mat[0].size();j++){
                col_sum[i]+=mat[j][i];
            }
            max_sum=max(max_sum,col_sum[i]);
        }
        int min_col_inc=0;
        int min_row_inc=0;
        for(int i=0;i<row_sum.size();i++){
            if(row_sum[i]<max_sum){
                min_row_inc+=max_sum-row_sum[i];
            }
        }
        for(int i=0;i<col_sum.size();i++){
            if(col_sum[i]<max_sum){
                min_col_inc+=max_sum-col_sum[i];
            }
        }
        return min(min_col_inc,min_row_inc);
    }
};